api_docs: Edit linked markdown files in endpoint descriptions.

laurynmm opened the pull request:
Small clean up of linked markdown files in the OpenAPI documentation. I noticed these in the api documentation when reviewing #22222.

Notes on changes:
- Moves linked markdown files to the end of the two impacted endpoint's descriptions (previously they were at the start of the description). This change becomes more relevant when the PR mentioned above is merged as it moves where the endpoint's method and path are rendered in the documentation.

- Adds a link to the help center documentation on user roles and permissions in `api-members-only.md`.

- Restructures warning in `can-create-users-only.md` so that it renders the information about the `can_create_users` permission as one paragraph and renders the **Changes** note as a separate warning.

Questions:
- Do we want the **Changes** note for the "Create a user" endpoint in a separate file? Or do we want to have that directly in the OpenAPI documentation? I believe it's in the linked file for the formatting of the warning, which highlights it in the page more than it would be normally.

---

**Screenshots and screen captures:**

<details>
<summary>Get user groups</summary>

[Link to current documentation](https://zulip.com/api/get-user-groups)

**Changes**:
![Screenshot from 2022-06-21 17-52-48](https://user-images.githubusercontent.com/63245456/174859126-c4610f00-3d67-4f00-bc3b-6d0e18f111d5.png)
</details>

<details>
<summary>Create a user</summary>

[Link to current documentation](https://zulip.com/api/create-user)

**Changes**:
![Screenshot from 2022-06-21 17-53-05](https://user-images.githubusercontent.com/63245456/174863235-76dd977c-74a9-4006-a17c-af4b83f45901.png)

</details>

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

timabbott commented:
> Do we want the Changes note for the "Create a user" endpoint in a separate file? Or do we want to have that directly in the OpenAPI documentation? I believe it's in the linked file for the formatting of the warning, which highlights it in the page more than it would be normally.

I think in an ideal world, we'd not have a separate include file that's only used in one place; I'm not entirely sure why `can-create-users-only.md` is a separate file, rather than being inlined in the documentation. @mateuszmandera do you recall?

laurynmm commented:
FYI - [This is the commit](https://github.com/zulip/zulip/commit/160cc5120aae0870c68e6e3a12a391035bd25843) when `why-create-users-only.md` was created. From that commit, it looks like the generic `api-admin-only.md` file used to be linked directly in the OpenAPI documentation, while now it's added in the `APITitlePreprocessor`. And `can-create-users-only.md` was a specialized version of that warning for this endpoint.

And [here's the commit](https://github.com/zulip/zulip/commit/8c9dd6198f60805c411222926ac6e5e3ccda7f31) where both files were created, where they were linked in the specific endpoint markdown files (which don't exist anymore). Note that `api-members-only.md` is also currently only being used for that one endpoint.

So, it seems to me the question here is do we want these single-use, linked files, which means the information is not directly in the documentation, but we get the warning block formatting in the rendered documentation? Or do we want to move this information and lose the warning formatting in the rendered docs?

mateuszmandera commented:
@timabbott 

> FYI - [This is the commit](https://github.com/zulip/zulip/commit/160cc5120aae0870c68e6e3a12a391035bd25843) when `why-create-users-only.md` was created. From that commit, it looks like the generic `api-admin-only.md` file used to be linked directly in the OpenAPI documentation, while now it's added in the `APITitlePreprocessor`. And `can-create-users-only.md` was a specialized version of that warning for this endpoint.

Yeah, that's right. And I think the reason for not inlining was that it seemed unnatural in `zulip.yaml`. I'm also not sure if the `!!warn` syntax even worked correctly if inlined there, I may have had some issues with it back then, but my memory isn't very clear on that.

If we think that just inlining it into `zulip.yaml` is fine in terms of style, and the ```!!warn` thing works correctly, then I'm not aware of any other good reason to keep it in a separate file.

laurynmm commented:
> If we think that just inlining it into `zulip.yaml` is fine in terms of style, and the `!!warn` thing works correctly, then I'm not aware of any other good reason to keep it in a separate file.

We do already use the `!!! warn` in the documentation in the endpoint description for [`fetch-api-key`](https://zulip.com/api/fetch-api-key), so it does work (though now I want to fix that to be a single paragraph).

Though, perhaps worth noting that it does look odd when the documentation is rendered in other OpenAPI documentation generators (follow the link in [this CZO message](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/Outdated.20API.20response.20doc.20in.20get_streams/near/1390699)).

timabbott commented:
> Though, perhaps worth noting that it does look odd when the documentation is rendered in other OpenAPI documentation generators

I think we should ignore that issue for now -- that issue applies to the include library in the first place, and so I think it's best to just do what is clean with our current infrastructure. (I imagine the way we'll support third-party OpenAPI documentation generators will likely involve running some sort of preprocessor on `zulip.yaml`, or potentially replacing this fancier tip syntax with raw HTML if that doesn't prove too ugly).

So I think it makes sense to inline this notice?

laurynmm commented:
Sounds good to me. This is updated for moving the warning notes to be inline text in the descriptions. I updated the commit message to reflect that, and I added a 2nd commit updating some other notes in the OpenAPI documentation that seemed important enough to also have the warning block formatting.

---

### 1st commit
- Removes the include links and puts the warning text inline in the documentation.
- Deletes both include markdown files because they are now unused.
- Cleans up the other existing warning block in `/fetch-api-key` endpoint.

**Screenshots**:
Note: I manually updated **Create user** and **Get user groups** screenshots for how they'll be when #22222 is merged to show why I moved the placement of the warning text to not be the first part of the description.

<details>
<summary>Create user</summary>

[Current documentation](https://zulip.com/api/create-user)
![Screenshot from 2022-07-08 16-12-51](https://user-images.githubusercontent.com/63245456/178017978-c02489fd-9aa4-4239-bfa7-bcceca240f99.png)
</details>

<details>
<summary>Get user groups</summary>

[Current documentation](https://zulip.com/api/get-user-groups)
![Screenshot from 2022-07-08 16-13-13](https://user-images.githubusercontent.com/63245456/178018019-63ba203b-8dce-4f16-ae91-5df117412b62.png)
</details>

<details>
<summary>Fetch API key (production)</summary>

[Current documentation](https://zulip.com/api/fetch-api-key)
![Screenshot from 2022-07-08 16-13-37](https://user-images.githubusercontent.com/63245456/178018057-1f2c3e53-7854-4095-93a7-d5260d9033c6.png)
</details>

---

### 2nd commit:
- Found two other bold **Notes** in endpoint descriptions that seemed to benefit from warning block formatting while we're making these updates

**Screenshots**:

<details>
<summary>Fetch API key (dev only)</summary>

[Current documentation](https://zulip.com/api/dev-fetch-api-key)
![Screenshot from 2022-07-08 16-44-12](https://user-images.githubusercontent.com/63245456/178019596-a9d700c8-9c6b-4d3e-bb3c-28c14ec62c93.png)
</details>

<details>
<summary>Get events: parameters</summary>

[Current documentation](https://zulip.com/api/get-events#parameters)
![Screenshot from 2022-07-08 16-43-49](https://user-images.githubusercontent.com/63245456/178019651-c10b58fc-13e9-478f-b144-44548cfed559.png)
</details>

timabbott commented:
Merged, thanks @laurynmm!
